### PR TITLE
Feature/24 add pluggin packages (verdaccio-htpasswd)

### PR DESCRIPTION
**Pull request by griffithtp**
**Type:** feature/build
**Scope:** plugins

The following has been addressed in the PR:
** verdaccio-htpasswd from verdaccio-htpasswd

*  There is a related issue? #24 
*  Unit or Functional tests are included in the PR - Yes, and untouched from existing repositories

**Description:**
~The import process for each individual plugin packages were 1) git clone, 2) remove .git, .gitignore, yarn.lock folder and files, 3) yarn and yarn build to validate.~
The suggested import process were to use `lerna import` in order to keep the tracking history and help with further bug hunting.
Also, a granular approach on one respective pluging ensures better control and support.
Build pipeline scripts are to be updated too.

### CODE REVIEW

**Comment by juanpicado** [comment]
awesome ... that was fast @griffithtp .

Could you leave `verdaccio-google-cloud` out ? I need to finish https://github.com/verdaccio/verdaccio-google-cloud/pull/20 this first.

**Comment by juanpicado** [comment]
Also `@verdaccio/local-storage`, we have many PR in progress. I'd wait until we finish that.

**Comment by griffithtp** [comment]
> awesome ... that was fast @griffithtp .
> 
> Could you leave `verdaccio-google-cloud` out ? I need to finish [verdaccio/verdaccio-google-cloud#20](https://github.com/verdaccio/verdaccio-google-cloud/pull/20) this first.

Yes roger that 😄 
It might actually be more beneficial to do one plugin PR at a time and I will try @sergiohgz recommended approach to use lerna import

**Comment by juanpicado** [comment]
Sure, let's wait @sergiohgz feedback.

**Comment by sergiohgz** [comment]
@griffithtp and I have talked about the migration process and it make more sense to make using `lerna import` command to keep track of all commits done in the older repo. Also, we talked about how to modernize project and that is better to make each plugin in a simple PR to make review faster and easier.

**Comment by griffithtp** [comment]
ok so this was interesting.
Even with the `--flatten` option it included the commit history which is good in a sense to enable to backtrack issues in future with VS Code gitlens extension for example?
If you prefer me to squash them let me know @sergiohgz @juanpicado 
Some notes/caveat for discussion:
* I had to keep .gitignore since the build output `lib` folder should not be included
* remove dependencies from the plugin/htpasswd folder that are already in the main package.json - perhaps separate PR, or I can do here too, especially if they are the same versions
* linting - consolidate into root folder?
* .yarnrc - remove from each of plugins' folder?
* docs - consolidate CODE_OF_CONDUCT and CONTRIBUTING from each of the plugins' folders and update plugin's README to link up to root's respective doc?

File: plugins/htpasswd/.github/FUNDING.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+open_collective: verdaccio
```
Comment (sergiohgz) [inline plugins/htpasswd/.github/FUNDING.yml]: `.github` folder is not necessary as the main project contains one

File: plugins/htpasswd/.gitignore [hunk 2]
```diff
@@ -0,0 +1,11 @@
+npm-debug.log
+.DS_Store
+package-lock.json
+_storage/
+lib/
```
Comment (sergiohgz) [inline plugins/htpasswd/.gitignore]: I agree that each project has own settings, so they can have their own `.gitignore` with the necessary rules they need to add (by example, lockfiles or node_modules are not necessary because the subproject doesn't contains own lockfiles and the node_modules are defined in an upper scope

File: plugins/htpasswd/.npmignore [hunk 3]
```diff
@@ -0,0 +1,25 @@
+src/
```
Comment (sergiohgz) [inline plugins/htpasswd/.npmignore]: I think this file is not needed

File: plugins/htpasswd/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "useTabs": false,
+  "printWidth": 80,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "trailingComma": "none",
+  "parser": "flow",
+  "semi": true
+}
```
Comment (sergiohgz) [inline plugins/htpasswd/.prettierrc]: Well, in next steps we should unify linting of all subproject (maybe when we import the rest), but it requires some discuss before, so this could stay as it is now

File: plugins/htpasswd/.yarnrc [hunk 5]
```diff
@@ -0,0 +1,2 @@
+save-prefix ""
+registry "http://registry.npmjs.org/"
```
Comment (sergiohgz) [inline plugins/htpasswd/.yarnrc]: Is this file necessary @juanpicado ?

File: plugins/htpasswd/CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Contributor Covenant Code of Conduct
```
Comment (sergiohgz) [inline plugins/htpasswd/CODE_OF_CONDUCT.md]: I think that in subprojects we do not need a CODE_OF_CONDUCT because it is in main project

File: plugins/htpasswd/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+# Contributing
```
Comment (sergiohgz) [inline plugins/htpasswd/CONTRIBUTING.md]: With CONTRIBUTING we can do 2 things: remove and improve outer CONTRIBUTING or refactor this one to be specific to the project. I prefer the first one, but what do you think @verdaccio/core-team and @griffithtp ?

File: plugins/htpasswd/LICENSE [hunk 8]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2017 Verdaccio
```
Comment (sergiohgz) [inline plugins/htpasswd/LICENSE]: Can you upgrade LICENSE year to current?

File: plugins/htpasswd/package.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    "codecov": "3.2.0",
+    "cross-env": "5.2.0",
+    "eslint": "5.15.3",
+    "get-stdin": "6.0.0",
+    "husky": "0.14.3",
+    "in-publish": "2.0.0",
+    "jest": "24.5.0",
+    "lint-staged": "7.2.0",
+    "standard-version": "5.0.2",
+    "typescript": "3.2.4"
```
Comment (sergiohgz) [inline plugins/htpasswd/package.json]: The devDependencies can be refactored after this PR. it's a big job to ensure everything works with the same versions :wink:

File: plugins/htpasswd/scripts/github-release.sh [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# Get the last tag from GitHub
+lastTag=$(git describe --tags $(git rev-list --tags --max-count=1))
+
+changelog=$(git show $GITHUB_SHA --unified=0 CHANGELOG.md | tail +12 | sed -e 's/^\+//')
```
Comment (sergiohgz) [inline plugins/htpasswd/scripts/github-release.sh]: I think the `scripts` folder is not needed because the scripts contained are based on a single project repository behaviour

File: plugins/htpasswd/.editorconfig [hunk 11]
```diff
@@ -0,0 +1,12 @@
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+# 2 space indentation
+[{.,}*.{js,yml,yaml}]
```
Comment (juanpicado) [inline plugins/htpasswd/.editorconfig]: I'd add here `ts`

File: plugins/htpasswd/.flowconfig [hunk 12]
```diff
@@ -0,0 +1,11 @@
+[ignore]
```
Comment (juanpicado) [inline plugins/htpasswd/.flowconfig]: I think we can remove `.flowconfig`. But let it for next PR.

File: plugins/htpasswd/.prettierrc (hunk 4 above)
Comment (juanpicado) [reply plugins/htpasswd/.prettierrc]: yes, I think some dot files are outdated, we can unify in a further step.

File: plugins/htpasswd/.yarnrc (hunk 5 above)
Comment (juanpicado) [reply plugins/htpasswd/.yarnrc]: Nop, we should remove it

File: plugins/htpasswd/CONTRIBUTING.md (hunk 7 above)
Comment (juanpicado) [reply plugins/htpasswd/CONTRIBUTING.md]: Yes, one CONTRIBUTING is enough.

File: plugins/htpasswd/scripts/github-release.sh (hunk 10 above)
Comment (juanpicado) [reply plugins/htpasswd/scripts/github-release.sh]: All GitHub hidden folder should be removed.

**Comment by juanpicado** [comment]
Agree with @sergiohgz suggestions above.

File: plugins/htpasswd/.editorconfig (hunk 11 above)
Comment (sergiohgz) [reply plugins/htpasswd/.editorconfig]: You are right with the config change, but I think this change should be done in other moment, to migrate as close as the original repo is

Comment (juanpicado) [reply plugins/htpasswd/.editorconfig]: agree

File: plugins/htpasswd/.gitignore (hunk 2 above)
Comment (griffithtp) [reply plugins/htpasswd/.gitignore]: what do you think about `dist/` folder for all the build outputs?

File: plugins/htpasswd/CONTRIBUTING.md (hunk 7 above)
Comment (griffithtp) [reply plugins/htpasswd/CONTRIBUTING.md]: one CONTRIBUTING, unless different packages has different specific contributing notes

**Comment by griffithtp** [comment]
btw @sergiohgz @juanpicado, I just noticed root .gitignore has tsconfig.json and jsconfig.json
we need tsconfig don't we?

**Comment by sergiohgz** [comment]
@griffithtp thanks for reporting, remove both, tsconfig is mandatory and jsconfig is optional

File: plugins/htpasswd/.gitignore (hunk 2 above)
Comment (sergiohgz) [reply plugins/htpasswd/.gitignore]: `build/` and `dist/` folders shouldn't go to the repository, so add both

**Comment by griffithtp** [comment]
@sergiohgz @juanpicado 
let me know if there is anything else to be done here?
I would be keen on continuing to importing the other plugins on separate PRs

**Comment by juanpicado** [comment]
@griffithtp could you resolve the conflict?
@sergiohgz I think this depends mostly of you.

**Comment by sergiohgz** [comment]
@griffith You have done a great job with the import, so I let you decide if you want to finish with the modernization of this plugin or prefer to start with other :wink:.
If you decide the first option, take a look on how to continue, take a look at [these commits](https://github.com/verdaccio/monorepo/pull/28/files/f4bbf3af09deaaea775a82b5205bc453f73704e9..cebb1035f8eb10593a32752de824cab73b8fe489). The most important things are:
- We have decided to remove `.npmignore`, the blacklist of what to pack into the package, so we have to whitelist `lib` folder using `files` property in `package.json`. Optionally, add the `types` property in `package.json` referencing `lib/index.d.ts`
- Remove old scripts (commitmsg, precommit and release), add `"lint:stage": "lint-staged"` script, sort alphabetically the rest
- Remove devDependencies that are already defined in the root package, and flatten the rest (except those that are packages defined in the monorepo, like `@verdaccio/eslint-config` or `@verdaccio/types` (this will be migrated sooner or later)). Also, add caret to devDependencies (but not dependencies).
- Add `"access": true` in `publishConfig` property (`package.json`) if the package is a scoped package (not in this case)

**Comment by juanpicado** [comment]
It make sense the new recommendations 👍

**Comment by juanpicado** [comment]
I already deprecated the repo https://github.com/verdaccio/verdaccio-htpasswd and archived. So nobody can create more PR.

**Comment by griffithtp** [comment]
> @Griffith You have done a great job with the import, so I let you decide if you want to finish with the modernization of this plugin or prefer to start with other 😉.
> If you decide the first option, take a look on how to continue, take a look at [these commits](https://github.com/verdaccio/monorepo/pull/28/files/f4bbf3af09deaaea775a82b5205bc453f73704e9..cebb1035f8eb10593a32752de824cab73b8fe489). The most important things are:
> 
> * We have decided to remove `.npmignore`, the blacklist of what to pack into the package, so we have to whitelist `lib` folder using `files` property in `package.json`. Optionally, add the `types` property in `package.json` referencing `lib/index.d.ts`
> * Remove old scripts (commitmsg, precommit and release), add `"lint:stage": "lint-staged"` script, sort alphabetically the rest
> * Remove devDependencies that are already defined in the root package, and flatten the rest (except those that are packages defined in the monorepo, like `@verdaccio/eslint-config` or `@verdaccio/types` (this will be migrated sooner or later)). Also, add caret to devDependencies (but not dependencies).
> * Add `"access": true` in `publishConfig` property (`package.json`) if the package is a scoped package (not in this case)

Thanks @sergiohgz 
Sorry I'm not sure I fully understand the access:true where you are asking to add it yet as you mention it's for scoped package which is not in this case. Do you need added here or is it not needed, is that what you mean? And btw, what does this setting do?

File: plugins/htpasswd/package.json [hunk 13]
```diff
@@ -34,29 +36,20 @@
...
+    "@verdaccio/eslint-config": "^0.0.1",
+    "cross-env": "^5.2.0",
+    "get-stdin": "^6.0.0",
+    "in-publish": "^2.0.0",
+    "standard-version": "^5.0.2"
   },
   "publishConfig": {
-    "registry": "https://registry.npmjs.org/"
+    "registry": "https://registry.npmjs.org/",
+    "access": true
```
Comment (juanpicado) [inline plugins/htpasswd/package.json]: @griffithtp here, I think we should remove the whole `publishConfig` section.

**Comment by sergiohgz** [comment]
@griffithtp First, sorry with the `publishConfig` property, there is a typo above I’m going to correct. The property `“access”` is not a boolean, it’s a string value, specifically it should be `“public”`.

You need a explanation on why and when we need that property. Take a look at [`lerna publish command docs`](https://github.com/lerna/lerna/tree/master/commands/publish#publishconfigaccess)

I am sorry because the typo and the bad explanation in the message before, I think that you have understood now 😉

File: plugins/htpasswd/package.json [hunk 14]
```diff
@@ -34,29 +36,20 @@
...
-    "cross-env": "5.2.0",
-    "eslint": "5.15.3",
-    "get-stdin": "6.0.0",
-    "husky": "0.14.3",
-    "in-publish": "2.0.0",
-    "jest": "24.5.0",
-    "lint-staged": "7.2.0",
-    "standard-version": "5.0.2",
-    "typescript": "3.2.4"
+    "@commitlint/config-conventional": "^7.5.0",
```
Comment (sergiohgz) [inline plugins/htpasswd/package.json]: We don’t need commitlint settings in a plugin package, it’s defined in the root :wink:

File: plugins/htpasswd/package.json [hunk 15]
```diff
@@ -34,29 +36,20 @@
...
+    "@octokit/rest": "^16.23.2",
+    "@types/bcryptjs": "^2.4.2",
+    "@verdaccio/babel-preset": "^0.1.0",
+    "@verdaccio/types": "^5.0.0-beta.4",
+    "@verdaccio/eslint-config": "^0.0.1",
+    "cross-env": "^5.2.0",
+    "get-stdin": "^6.0.0",
+    "in-publish": "^2.0.0",
+    "standard-version": "^5.0.2"
```
Comment (sergiohgz) [inline plugins/htpasswd/package.json]: We don’t need standard-version, releasing is managed by lerna

File: plugins/htpasswd/package.json [hunk 16]
```diff
@@ -34,29 +36,20 @@
...
-    "jest": "24.5.0",
-    "lint-staged": "7.2.0",
-    "standard-version": "5.0.2",
-    "typescript": "3.2.4"
+    "@commitlint/config-conventional": "^7.5.0",
+    "@octokit/rest": "^16.23.2",
+    "@types/bcryptjs": "^2.4.2",
+    "@verdaccio/babel-preset": "^0.1.0",
+    "@verdaccio/types": "^5.0.0-beta.4",
+    "@verdaccio/eslint-config": "^0.0.1",
```
Comment (sergiohgz) [inline plugins/htpasswd/package.json]: These 3 @verdaccio dependencies can be updated to
```json
“@verdaccio/babel-preset”: “^2.0.0”,
“@verdaccio/eslint-config”: “^2.0.0”,
“@verdaccio/types”: “^7.0.0”,
```

File: plugins/htpasswd/package.json [hunk 17]
```diff
@@ -34,29 +36,20 @@
...
+    "cross-env": "^5.2.0",
+    "get-stdin": "^6.0.0",
+    "in-publish": "^2.0.0",
```
Comment (sergiohgz) [inline plugins/htpasswd/package.json]: Can you flatten the rest devDependencies to root package.json?

File: plugins/htpasswd/package.json (hunk 13 above)
Comment (sergiohgz) [reply plugins/htpasswd/package.json]: In this case, yes, remove whole publishConfig, we don’t need it

**Comment by juanpicado** [comment]
@sergiohgz about the `publishConfig` prop, I'd recommend do not use it, actually it cannot be overwritten by `--registry` or `.npmrc`.

File: plugins/htpasswd/package.json (hunk 17 above)
Comment (griffithtp) [reply plugins/htpasswd/package.json]: not entirely sure what it means by flatten devDependencies, sorry guys

**Comment by griffithtp** [comment]
@sergiohgz @juanpicado 
hey guys, yarn.lock keeps on conflicting everytime other PR are merged before this one while this has been open for a week almost now.
what do you recommend we do here?

**Comment by sergiohgz** [comment]
That is, merge master into your current branch and fix yarn.lock by performing yarn install :wink:

**Comment by griffithtp** [comment]
> That is, merge master into your current branch and fix yarn.lock by performing yarn install 😉

Thanks @sergiohgz, I believe I have done that but looking at the build failing, it appears some deps are missing that were required in the core/types?

**Comment by juanpicado** [comment]
`"@types/node"` is missing ...
```
@verdaccio/streams: > tsc --emitDeclarationOnly |  
-- | --
  | @verdaccio/readme: Successfully compiled 1 file with Babel. |  
  | verdaccio-htpasswd: $ npm run build:types && npm run build:js |  
  | verdaccio-htpasswd: npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1564318097716-0.018047426947226386/node but npm is using /opt/hostedtoolcache/node/12.7.0/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with. |  
  | verdaccio-htpasswd: > verdaccio-htpasswd@2.0.0 build:types /home/vsts/work/1/s/plugins/htpasswd |  
  | verdaccio-htpasswd: > tsc --emitDeclarationOnly |  
  | verdaccio-htpasswd: ../../core/types/index.d.ts(1,37): error TS2307: Cannot find module 'stream'. |  
  | verdaccio-htpasswd: ../../core/types/index.d.ts(8,29): error TS2503: Cannot find namespace 'NodeJS'. |  
  | verdaccio-htpasswd: ../../core/types/index.d.ts(427,19): error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node`. |  
  | verdaccio-htpasswd: ../../core/types/index.d.ts(427,28): error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node`. |  
  | verdaccio-htpasswd: src/crypt3.ts(10,20): error TS2307: Cannot find module 'crypto'. |  
  | verdaccio-htpasswd: src/htpasswd.ts(1,16): error TS2307: Cannot find module 'fs'. |  
  | verdaccio-htpasswd: src/htpasswd.ts(2,18): error TS2307: Cannot find module 'path'. |  
  | verdaccio-htpasswd: src/utils.ts(1,20): error TS2307: Cannot find module 'crypto'. |  
  | verdaccio-htpasswd: npm ERR! code ELIFECYCLE |  
  | verdaccio-htpasswd: npm ERR! errno 2 |  
  | verdaccio-htpasswd: npm ERR! verdaccio-htpasswd@2.0.0 build:types: `tsc --emitDeclarationOnly` |  
  | verdaccio-htpasswd: npm ERR! Exit status 2 |  
  | verdaccio-htpasswd: npm ERR! |  
  | verdaccio-htpasswd: npm ERR! Failed at the verdaccio-htpasswd@2.0.0 build:types script. |  
  | verdaccio-htpasswd: npm ERR! This is probably not a problem with npm. There is likely additional logging output above. |  
  | @verdaccio/streams: npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1564318096714-0.367587393640336/node but npm is using /opt/hostedtoolcache/node/12.7.0/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
```

**Comment by griffithtp** [comment]
> `"@types/node"` is missing ... @sergiohgz @griffithtp
> 
> ```
> @verdaccio/streams: > tsc --emitDeclarationOnly |  
> -- | --
>   | @verdaccio/readme: Successfully compiled 1 file with Babel. |  
>   | verdaccio-htpasswd: $ npm run build:types && npm run build:js |  
>   | verdaccio-htpasswd: npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1564318097716-0.018047426947226386/node but npm is using /opt/hostedtoolcache/node/12.7.0/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with. |  
>   | verdaccio-htpasswd: > verdaccio-htpasswd@2.0.0 build:types /home/vsts/work/1/s/plugins/htpasswd |  
>   | verdaccio-htpasswd: > tsc --emitDeclarationOnly |  
>   | verdaccio-htpasswd: ../../core/types/index.d.ts(1,37): error TS2307: Cannot find module 'stream'. |  
>   | verdaccio-htpasswd: ../../core/types/index.d.ts(8,29): error TS2503: Cannot find namespace 'NodeJS'. |  
>   | verdaccio-htpasswd: ../../core/types/index.d.ts(427,19): error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node`. |  
>   | verdaccio-htpasswd: ../../core/types/index.d.ts(427,28): error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node`. |  
>   | verdaccio-htpasswd: src/crypt3.ts(10,20): error TS2307: Cannot find module 'crypto'. |  
>   | verdaccio-htpasswd: src/htpasswd.ts(1,16): error TS2307: Cannot find module 'fs'. |  
>   | verdaccio-htpasswd: src/htpasswd.ts(2,18): error TS2307: Cannot find module 'path'. |  
>   | verdaccio-htpasswd: src/utils.ts(1,20): error TS2307: Cannot find module 'crypto'. |  
>   | verdaccio-htpasswd: npm ERR! code ELIFECYCLE |  
>   | verdaccio-htpasswd: npm ERR! errno 2 |  
>   | verdaccio-htpasswd: npm ERR! verdaccio-htpasswd@2.0.0 build:types: `tsc --emitDeclarationOnly` |  
>   | verdaccio-htpasswd: npm ERR! Exit status 2 |  
>   | verdaccio-htpasswd: npm ERR! |  
>   | verdaccio-htpasswd: npm ERR! Failed at the verdaccio-htpasswd@2.0.0 build:types script. |  
>   | verdaccio-htpasswd: npm ERR! This is probably not a problem with npm. There is likely additional logging output above. |  
>   | @verdaccio/streams: npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1564318096714-0.367587393640336/node but npm is using /opt/hostedtoolcache/node/12.7.0/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
> ```

Saw that in CircleCI but not sure how that came across considering https://github.com/verdaccio/monorepo/blob/master/package.json#L17

**Comment by juanpicado** [comment]
@griffithtp ok you are right, that deserves some research, but in order to do not block you, let's add them, we will figure out later on what's the reason of it. cc: @sergiohgz

File: plugins/htpasswd/package.json (hunk 17 above)
Comment (sergiohgz) [reply plugins/htpasswd/package.json]: I know this is complex to understand, we are improving CONTRIBUTING to add information about how to work with, create packages and importing packages. I will ask you to review if something is missing in this future CONTRIBUTING :wink:

With flatten devDependencies I mean to move devDependencies, that those code is not from this codebase, to the root package, so we can manage them in a single place and share between projects. 
By example, we cannot flatten `@verdaccio/eslint-config` because is a package in the codebase and **Yarn Workspaces** needs it defined in each workspace to link it correctly, but `in-publish` is a common devDependency that can be moved to the root `package.json` and manage one package version for every package that needs to use it.

**Comment by sergiohgz** [comment]
@juanpicado @griffithtp I'm agree, we can review the error later, let's not get it blocked  :wink:

**Comment by griffithtp** [comment]
it's already added, thus me linking to it in my previous comment.
@sergiohgz @juanpicado could you take a look at circleci or would you have any ideas why it would fail?

**Comment by sergiohgz** [comment]
@griffithtp I have fixed the problem with the correct path to types in typeRoots settings and also I fixed prettier settings to run on Windows. If the CI passes, I think we will merge this great job you did.

**Comment by griffithtp** [comment]
> 🏅👏👏👏👏👏
> 
> I need to say thank you because you did a really great job with this PRs, not only importing the package. You give us some clues of what we need to document better to understand how to work with monorepos, how to import packages, and more.

guys, thank you for guidance and for being patient with this PR 😄
